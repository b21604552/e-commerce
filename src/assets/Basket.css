.basket-wrapper {
    width: 120px;
    height: 48px;
    margin-right: 80px;
}

.basket-button {
    width: 120px;
    height: 48px;
    border: 1px solid #B0B0B0;
    box-sizing: border-box;
    border-radius: 4px;
    font-style: normal;
    font-weight: normal;
    font-size: 17px;
    line-height: 24px;
    display: flex;
    align-items: center;
    text-align: center;
    justify-content: center;
    color: #B0B0B0;
    cursor: pointer;
}

.basket-wrapper:hover>.basket-button {
    height: 47px;
    border-bottom: 0;
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
}

.basket-opened {
    display: none;
    width: 360px;
    border-radius: 4px;
    border: 1px solid #B0B0B0;
    box-sizing: border-box;
    z-index: 10000;
    position: absolute;
    margin-left: -240px;
    border-top-right-radius: 0;
    border-top: none;
    background: #FFFFFF linear-gradient(#B0B0B0, #B0B0B0) no-repeat top left;
    background-size: 240px 1px;
    padding: 21px;
}

.basket-wrapper:hover>.basket-opened {
    display: block;
}

.basket-count {
    z-index: 1000;
    position: absolute;
    border-radius: 25px;
    width: 18px;
    height: 18px;
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 12px;
    display: flex;
    align-items: center;
    justify-content: center;
    color: #FFFFFF;
    background: #FF6000;
    margin-left: 108px;
    margin-top: -7px;
}

.basket-opened-element {
    display: flex;
    margin-bottom: 20px;
}

.basket-product-image-wrapper {
    width: 54px;
    height: 72px;
    margin-right: 10px;
}

.basket-product-info-wrapper {

}

.basket-product-title {
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 18px;
    color: #484848;
    height: 48px;
}

.basket-remove-button {
    background: #FFFFFF;
    border: 1px solid #F90000;
    border-radius: 4px;
    font-style: normal;
    font-weight: normal;
    font-size: 10px;
    line-height: 18px;
    display: flex;
    align-items: center;
    text-align: center;
    color: #F90000;
    cursor: pointer;
}

.basket-empty {
    font-style: normal;
    font-weight: normal;
    font-size: 12px;
    line-height: 18px;
    color: #484848;
}

.basket-empty-wrapper {
    display: flex;
    justify-content: center;
    align-items: center;
}