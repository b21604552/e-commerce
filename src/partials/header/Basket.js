import React, { useEffect, useState } from 'react'
import Modal from '../modal/Modal'
import Api from '../../services/Api'
import GeneralHelper from '../../helper/GeneralHelper'
import { useSelector } from 'react-redux'
import { setModalVisible } from '../../services/redux/actions/GeneralAction'

export default function Basket () {
  const { itemCountInBasket } = useSelector((state) => state?.GeneralReducer)
  const { modalFinished } = useSelector((state) => state?.GeneralReducer)

  const [basket, setBasket] = useState(Api.getProductsInBasket())
  const [removingProduct, setRemovingProduct] = useState(null)

  useEffect(() => {
    if (modalFinished) {
      Api.removeProductToBasket(removingProduct)
    }
  }, [modalFinished, removingProduct])

  useEffect(() => {
    setBasket(Api.getProductsInBasket())
  }, [itemCountInBasket])

  function removeProductModal (id) {
    setRemovingProduct(id)
    setModalVisible(true)
  }

  return (
    <>
      <Modal/>
      <div className={'basket-wrapper'}>
        {
          itemCountInBasket > 0 && (
            <i className="basket-count">{itemCountInBasket}</i>
          )
        }
        <div className="basket-button">
          <div>Sepetim</div>
        </div>
        <div className={'basket-opened'}>
          {basket.length !== 0 ? (
            basket.map((product) => (
              <div key={'product-' + product.id} className={'basket-opened-element'}>
                <div className={'basket-product-image-wrapper'}>
                  <img alt={'product-' + product.id} src={GeneralHelper.getImageByKey(product.image)} width={40} height={59}/>
                </div>
                <div className={'basket-product-info-wrapper'}>
                  <div className={'basket-product-title'}>
                    {product.title + ' Renk: ' + product.color}
                  </div>
                  <div className={'basket-product-remove-wrapper'}>
                    <button id={'remove-' + product.id} className={'basket-remove-button'} onClick={() => removeProductModal(product.id)}>Kaldır</button>
                  </div>
                </div>
              </div>
            ))
          ) : (
            <div className={'basket-empty-wrapper'}>
              <div className={'basket-empty'}>
                Sepetinizde ürün bulunmamaktadır.
              </div>
            </div>
          )
          }
        </div>
      </div>
    </>
  )
}