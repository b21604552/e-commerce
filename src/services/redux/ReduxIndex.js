import { createStore } from 'redux'
import reducerCombined from './reducerCombined'

export const reduxStore = createStore(reducerCombined)
