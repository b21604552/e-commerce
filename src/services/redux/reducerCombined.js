import { combineReducers } from 'redux'
import GeneralReducer from './reducers/GeneralReducer'

const reducerCombined = combineReducers({
  GeneralReducer: GeneralReducer
})

export default reducerCombined
