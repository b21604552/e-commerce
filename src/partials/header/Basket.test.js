import React from 'react'
import ReactDOM from 'react-dom'
import Basket from './Basket'
import renderer from 'react-test-renderer'
import { reduxStore } from '../../services/redux/ReduxIndex'
import { Provider } from 'react-redux'

describe('Basket Test', () => {
  let page = (
    <Provider store={reduxStore}>
      <Basket/>
    </Provider>
  )

  it('Renders Basket without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(page, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders Basket correctly', () => {
    const div = renderer.create(page).toJSON()
    expect(div).toMatchSnapshot()
  })
})