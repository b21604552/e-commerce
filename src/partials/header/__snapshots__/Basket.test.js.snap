// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basket Test Renders Basket correctly 1`] = `
<div
  className="basket-wrapper"
>
  <div
    className="basket-button"
  >
    <div>
      Sepetim
    </div>
  </div>
  <div
    className="basket-opened"
  >
    <div
      className="basket-empty-wrapper"
    >
      <div
        className="basket-empty"
      >
        Sepetinizde ürün bulunmamaktadır.
      </div>
    </div>
  </div>
</div>
`;
